File: CYBaseProject/CYBaseProject/Code/Utils/UI/UIImage+CYUtils.m
//
//  UIImage+EUUtils.m
//  Euaaaios
//
//  Created by HuangQiSheng on 4/14/16.
//  Copyright © 2016 euaaa. All rights reserved.
//

#import "UIImage+CYUtils.h"

@implementation UIImage (CYUtils)

- (UIImage *)imageByScaleImageToSize:(CGSize)size {
    
    UIGraphicsBeginImageContextWithOptions(size, NO, 0.0);
    [self drawInRect:CGRectMake(0, 0, size.width, size.height)];
    UIImage *newImage = UIGraphicsGetImageFromCurrentImageContext();
    UIGraphicsEndImageContext();
    return newImage;
}

- (UIImage *)imageByScaleImageWidthTo:(CGFloat)width {
    
    CGSize originSize = self.size;
    
    CGSize newSize = CGSizeMake(width, width * originSize.height / originSize.width);
    
    return [self imageByScaleImageToSize:newSize];
}

- (UIImage *)imageByScaleImageHeightTo:(CGFloat)height {
    
    CGSize originSize = self.size;
    
    CGSize newSize = CGSizeMake(height * originSize.width / originSize.height, height);
    return [self imageByScaleImageToSize:newSize];
}

@end

File: CYBaseProjectSwift/CYBaseProjectSwift/Code/Utils/UI/UIImage+CYUtils.swift
//
//  UIImage+CYUtils.swift
//  CYBaseProjectSwift
//
//  Created by HuangQiSheng on 4/22/16.
//  Copyright © 2016 Jasper. All rights reserved.
//

import UIKit

extension UIImage {
    
    func imageByScaleImageToSize(size: CGSize) -> UIImage {
        
        UIGraphicsBeginImageContextWithOptions(size, false, 0.0)
        self.drawInRect(CGRect(origin: CGPoint(x: 0, y: 0), size: size))
        let image = UIGraphicsGetImageFromCurrentImageContext()
        UIGraphicsEndImageContext()
        
        return image
    }
    
    func imageByScaleImageWidth(toWidth: CGFloat) -> UIImage {
        
        let originSize = self.size
        let newSize = CGSize(width: toWidth, height: originSize.height)
        
        return self.imageByScaleImageToSize(newSize)
    }
    
    func imageByScaleImageHeight(toHeight: CGFloat) -> UIImage {
        
        let originSize = self.size
        let newSize = CGSize(width: originSize.width, height: toHeight)
        
        return self.imageByScaleImageToSize(newSize)
    }
    
    class func resizableImage(named name: String, capInsets: UIEdgeInsets) -> UIImage? {
        
        return UIImage(named: name)?.resizableImageWithCapInsets(capInsets)
    }
    
}
